### PR TITLE
PLAT-838-2 Test for SystemEventNotifier

### PR DESCRIPTION
**Pull request by daniel-klose**

### CODE REVIEW

File: platform-core-module/src/test/java/com/softicar/platform/core/module/event/SystemEventNotifierTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+	public SystemEventNotifierTest() {
+
+		this.moduleInstance = AGCoreModuleInstance.getInstance();
+	}
+
+	@Test(expected = SofticarUserException.class)
+	public void testWithoutRecipients() {
+
+		SystemEventNotifier.notifyAboutEvents();
```
Comment (oliver-richers) [inline platform-core-module/src/test/java/com/softicar/platform/core/module/event/SystemEventNotifierTest.java:25]: IMO, should rather check the message with `assertException`

File: platform-core-module/src/test/java/com/softicar/platform/core/module/event/SystemEventNotifierTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	}
+
+	@Test
+	public void testWithoutRecipientsAndTestSystem() {
+
+		moduleInstance.setTestSystem(true).save();
+		SystemEventNotifier.notifyAboutEvents();
+		assertEquals(0, AGBufferedEmail.TABLE.countAll());
```
Comment (oliver-richers) [inline platform-core-module/src/test/java/com/softicar/platform/core/module/event/SystemEventNotifierTest.java:33]: hmm, you didn't add any events, so there are 2 reasons why nothing is sent

File: platform-core-module/src/test/java/com/softicar/platform/core/module/event/SystemEventNotifierTest.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	}
+
+	@Test
+	public void testWithEvents() {
+
+		insertSystemEventEmailRecipient();
+		new SystemEventBuilder(AGSystemEventSeverityEnum.ERROR, "Test").save();
+		SystemEventNotifier.notifyAboutEvents();
+		List<AGBufferedEmail> bufferedMails = AGBufferedEmail.TABLE.loadAll();
```
Comment (oliver-richers) [inline platform-core-module/src/test/java/com/softicar/platform/core/module/event/SystemEventNotifierTest.java:50]: we should separate test setup, test execution and test assertion here to make test easier to grasp
```suggestion

		SystemEventNotifier.notifyAboutEvents();

		var bufferedMails = AGBufferedEmail.TABLE.loadAll();
```

File: platform-core-module/src/test/java/com/softicar/platform/core/module/event/SystemEventNotifierTest.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+		insertSystemEventEmailRecipient();
+		SystemEventNotifier.notifyAboutEvents();
+		assertEquals(0, AGBufferedEmail.TABLE.countAll());
+	}
+
+	@Test
+	public void testWithEvents() {
+
+		insertSystemEventEmailRecipient();
+		new SystemEventBuilder(AGSystemEventSeverityEnum.ERROR, "Test").save();
```
Comment (oliver-richers) [inline platform-core-module/src/test/java/com/softicar/platform/core/module/event/SystemEventNotifierTest.java:48]: IMO should also be extracted to private method
```suggestion
		insertSystemEvent(AGSystemEventSeverityEnum.ERROR, "Test");
```

**Review by oliver-richers — CHANGES_REQUESTED** [review]

File: platform-core-module/src/test/java/com/softicar/platform/core/module/event/SystemEventNotifierTest.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+	@Test(expected = SofticarUserException.class)
+	public void testWithoutRecipients() {
+
+		SystemEventNotifier.notifyAboutEvents();
+	}
+
+	@Test
+	public void testWithoutRecipientsAndTestSystem() {
+
+		moduleInstance.setTestSystem(true).save();
```
[the remark this answers is not in the document]
Comment (daniel-klose) [reply platform-core-module/src/test/java/com/softicar/platform/core/module/event/SystemEventNotifierTest.java:31]: yes

**Review by oliver-richers — APPROVED** [review]

**Pull request merged**